sources/main.cpp

```diff
@@ -204,7 +205,24 @@ int main(int argc, char *argv[]) {
   boost::filesystem::path contingencyPath(runtimeConfig.contingenciesFilePath);
   if (!runtimeConfig.zipArchivePath.empty()) {
     if (mpiContext.isRootProc()) {
-      boost::shared_ptr<zip::ZipFile> archive = zip::ZipInputStream::read(runtimeConfig.zipArchivePath);
+      boost::shared_ptr<zip::ZipFile> archive;
+      try {
+        archive = zip::ZipInputStream::read(runtimeConfig.zipArchivePath);
+      } catch (const zip::ZipException& e) {
+        switch (e.getErrorCode()) {
+          case zip::Error::Code::LIBARCHIVE_INTERNAL_ERROR:
+            std::cerr << e.what() << " for the " << runtimeConfig.zipArchivePath << " archive" << std::endl;
```

Comment:
Can we add this in the dictionary?

Reply:
So we totally override the libzip error in this case ? We get rid of e.what() in the code.

Reply:
You can add something like "Zip error for file ... : e.what()" for example

Reply:
yes, I'm just proposing to use the dictionnary with e.what() as parameter of the message